digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/pointasset/OracleTrafficLightDao.scala

```diff
@@ -114,15 +140,26 @@ object OracleTrafficLightDao {
       val propertyPublicId = r.nextString
       val propertyType = r.nextString
       val propertyRequired = r.nextBoolean
-      val propertyValue = r.nextLongOption()
+      val propertyValue = r.nextDoubleOption()
       val propertyDisplayValue = r.nextStringOption()
+      val propertyGroupedId = r.nextLong
       val property = PropertyRow(
         propertyId = propertyId,
         publicId = propertyPublicId,
         propertyType = propertyType,
         propertyRequired = propertyRequired,
-        propertyValue = propertyValue.getOrElse(propertyDisplayValue.getOrElse("")).toString,
-        propertyDisplayValue = propertyDisplayValue.orNull)
+        propertyValue = propertyValue match {
+          case Some(double) =>
+            val value: AnyVal = if (double % 1 == 0) double.toLong else double
```

Comment:
Why we perform this and then a transformation to the value to String type?

Reply:
This validation is here to make a difference when representing values like 4.0 to 4 since now we have to treat this parameter as a Double type instead of an Int